### PR TITLE
Stable Docker setup

### PR DESCRIPTION
**Pull request by dnsl48**
Docker shim scripts refactoring, add stable and release setup, github action for a github package build and publication.

Resolves: https://github.com/silverstripe/cow/issues/146 https://github.com/silverstripe/cow/issues/163

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -2,45 +2,103 @@
...
 
-## ./docker/test
+That implies that Docker installed and running on the host machine becomes the only 3rd party
+requirement of COW. Not even PHP is required anymore since it's included into the docker images.
 
-This script runs all the unit tests within a container
+Containers are very cheap and disposable and we shouldn't think of them as of classic virtual machines (which
+are very expensive to initialize and spin up). As such, every script run creates a new
+container, runs its function and deletes the container afterwards. That means the only shared
+environment between them is your host machine filesystem (and environment variables).
```
[the remark this answers is not in the document]
Comment (dnsl48) [reply docker/README.md]: added at the top of the `Scripts` section

**Comment by dnsl48** [comment]
Addressed the feedback and discussed in person

**Comment by ScopeyNZ** [comment]
Hey team, quick question. I noticed the CI for this is GitHub actions. Is there a reason for this over Travis? Are we going to be moving more projects to actions rather than travis? I feel a bit out of the loop here.

**Comment by dnsl48** [comment]
> Hey team, quick question. I noticed the CI for this is GitHub actions. Is there a reason for this over Travis? Are we going to be moving more projects to actions rather than travis? I feel a bit out of the loop here.

Fair question! Yes, there is a reason for GH Actions over Travis, but particularly for publishing to GH Packages. We don't move CI away from Travis. The main reason is the close integration between GitHub, GitHub Actions and GitHub Packages so that all three reuse the same permission model and when you have read/write access to this repository you already have access to actions and packages. This eliminates any need to manage extra users (as it would be with DockerHub, or a machine user on to publish from Travis to GH Packages).

TL;DR; we only use GH Actions + GH Packages for building, storing and distributing the docker image and not using it for CI (not replacing Travis).
